### ISSUE TITLE
Problems mapping customized tree levels In Batch Editor and WB

### ISSUE THREAD

**Issue by timo11** [specify/specify6#803]
If a storage tree defition (probably any tree definition) contains customized levels with rankids that are not recognized by the default workbench and batch edit configuration files, even if the configuration files are modified correctly, if the custom columns' names or titles contain spaces, slashes or commas, or if their names are not the same as their titles, the custom columns will not be mappable, so they will not be batch-editable, and probably will not be uploadable either.

**Comment by timo11** [comment]
It is no longer necessary to avoid spaces or other white-spacey punctuationish characters, and names and no longer need to be the same as titles.

However, the workbench configuration files still need to be modified by hand to accommodate custom tree levels.
If a storage tree contains a rank named Case/Shelf/Tower with a rankid (which must be determined by querying the storagetreedefitem table) of 375, the following line would need to be inserted in preparation_update_wb_datamodel.xml:  
`<field column="CaseShelfTower1" name="case/Shelf/Tower1" type="java.lang.String" />`,
and the line below to specify_datamodel_workbench_upload_def.xml:
`    <field table="Preparation" name="case/Shelf/Tower1" onetomanysequence="0" treename="storage" maptofield="name" rankid="375"/>
`

**Comment by yvonneekanim** [comment]
@timo11 is this supposed to be case sensitive?

**Comment by timo11** [comment]
@yvonneekanim Yes, I am pretty sure the name in *datamodel.xml needs to exactly match the name in the upload_def file. Not sure for treename, maptofield and others.
